src/power_grid_model/validation/validation.py

```diff
@@ -221,6 +221,23 @@ def validate_ids_exist(update_data: Dict[str, np.ndarray], input_data: SingleDat
     return list(chain(*errors))
 
 
+def process_power_sigma_and_p_q_sigma(data: SingleDataset, sensor: str) -> None:
+    """
+    Helper function to process the required list when both `p_sigma` and `q_sigma` exist
+    and valid but `power_sigma` is missing. The field `power_sigma` is set to 1 in this case.
+    The argument is that when both `p_sigma` and `q_sigma` are valid, the value pf `power_sigma`
+    is always ignored.
+    """
+    if sensor in data and isinstance(data[sensor], np.ndarray):
+        sensor_data = data[sensor]
+        for i in range(len(sensor_data)):
```

Comment:
is there a way to do this using `np.logical_*` or `np.vectorize` to avoid the Python `for`-loop? or is that being done in plenty of other locations in the code already?

Reply:
Included in the latest commit.